apiserver/facades/agent/caasoperator/watcher.go

```diff
@@ -0,0 +1,108 @@
+// Copyright 2019 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package caasoperator
+
+import (
+	"github.com/juju/errors"
+	"gopkg.in/juju/worker.v1"
+	"gopkg.in/juju/worker.v1/catacomb"
+
+	corewatcher "github.com/juju/juju/core/watcher"
+	"github.com/juju/juju/state"
+)
+
+type unitIDWatcher struct {
+	catacomb catacomb.Catacomb
+	out      chan []string
+	src      corewatcher.StringsWatcher
+	model    watcherModelInterface
+}
+
+type watcherModelInterface interface {
+	Containers(providerIds ...string) ([]state.CloudContainer, error)
+}
+
+func newUnitIDWatcher(model watcherModelInterface, src corewatcher.StringsWatcher) (*unitIDWatcher, error) {
```

Comment:
even though not exported, warrants a comment to explain what this watcher does, what it emits etc